csrc/mmdeploy/backend_ops/tensorrt/common_impl/nms/allClassNMS.cu

```diff
@@ -66,17 +68,19 @@ __device__ float jaccardOverlap(const Bbox<T_BBOX> &bbox1, const Bbox<T_BBOX> &b
 /********** new NMS for only score and index array **********/
 
 template <typename T_SCORE, typename T_BBOX, int TSIZE>
-__global__ void allClassNMS_kernel(const int num, const int num_classes,
-                                   const int num_preds_per_class, const int top_k,
-                                   const float nms_threshold, const bool share_location,
-                                   const bool isNormalized,
-                                   T_BBOX *bbox_data,  // bbox_data should be float to preserve
-                                                       // location information
-                                   T_SCORE *beforeNMS_scores, int *beforeNMS_index_array,
-                                   T_SCORE *afterNMS_scores, int *afterNMS_index_array,
-                                   bool flipXY = false) {
+__global__ void
+#if __CUDA_ARCH__ == 620 || __CUDA_ARCH__ == 530
```

Comment:
Is it possible that `__CUDA_ARCH__` is undefined?

Reply:
NO as long as we compile cuda with nvcc. https://docs.nvidia.com/cuda/cuda-compiler-driver-nvcc/index.html#virtual-architecture-macros

Reply:
I used the codes [here](https://forums.developer.nvidia.com/t/cuda-arch-undefined/26274#nvcc-gencodearchcompute_20codesm_20compute_20-testcu-o-test-1) and compiled it with `nvcc -gencode arch=compute_50,code=sm_50 test.cu -o test`. The log gave me this: *test.cu(11): error: identifier "__CUDA_ARCH__" is undefined*

Reply:
> I used the codes [here](https://forums.developer.nvidia.com/t/cuda-arch-undefined/26274#nvcc-gencodearchcompute_20codesm_20compute_20-testcu-o-test-1) and compiled it with `nvcc -gencode arch=compute_50,code=sm_50 test.cu -o test`. The log gave me this: _test.cu(11): error: identifier "**CUDA_ARCH**" is undefined_

I found that the codes used `__CUDA_ARCH__` inside the func. If I placed it outside, there would be no error.

Reply:
> This macro can be used in the implementation of GPU functions for determining the virtual architecture for which it is currently being compiled. The host code (the non-GPU code) must not depend on it.

Host code would be passed to gcc or clang, which does not has the arch macro. Only device code would be compiled by nvcc.
try compile the code below:
```
#ifdef __CUDA_ARCH__
#warning "hello"
#endif
```